kong/db/dao/init.lua

```diff
@@ -1297,6 +1299,15 @@ function DAO:delete(primary_key, options)
   if not entity then
     return nil, tostring(err_t), err_t
   end
+  -- cascade entities are cleaned up by the strategy but we still need to decrement their counters
+  for _, entry in ipairs(cascade_entries) do
+    local dao = entry.dao
+    local cascade_entity = entry.entity
+    cascade_entity, err_t = run_hook("dao:delete:post", cascade_entity, dao.schema.name, options, cascade_entity.ws_id)
+    if not cascade_entity then
+      return nil, tostring(err_t), err_t
+    end
+  end
```

Comment:
this is the same as lines 912-920, right?  can it be factored into a function?

Reply:
Done